llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8294,6 +8294,17 @@ static std::optional<bool> isImpliedCondICmps(const ICmpInst *LHS,
   if (areMatchingOperands(L0, L1, R0, R1, AreSwappedOps))
     return isImpliedCondMatchingOperands(LPred, RPred, AreSwappedOps);
 
+  if (LPred == ICmpInst::ICMP_SGT) {
```

Comment:
Should also handle the less-than variant no?

Reply:
hi @goldsteinn
I reverted the change for less-than because the commit https://github.com/llvm/llvm-project/pull/66668#pullrequestreview-1682554666, so I think it may be addressed with some other method.
If you don't object, I think it is fine to keep this PR to fix the issue in the comment link.